share/completions/terraform.fish

```diff
@@ -1,178 +1,205 @@
+# Returns 0 if the command has not had a subcommand yet
+# Does not currently account for -chdir
+function __fish_terraform_needs_command
+    set -l cmd (commandline -opc)
+
+    if test (count $cmd) -eq 1
+        return 0
+    end
+
+    return 1
+end
+
 # general options
-complete -f -c terraform -l version -d 'Print version information'
-complete -f -c terraform -l help -d 'Show help'
-
-### apply
-complete -f -c terraform -n __fish_use_subcommand -a apply -d 'Build or change infrastructure'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o backup -d 'Path to backup the existing state file'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o lock -d 'Lock the state file when locking is supported'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o lock-timeout -d 'Duration to retry a state lock'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o input -d 'Ask for input for variables if not directly set'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o no-color -d 'If specified, output won\'t contain any color'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o parallelism -d 'Limit the number of concurrent operations'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o refresh -d 'Update state prior to checking for differences'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o state -d 'Path to a Terraform state file'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o state-out -d 'Path to write state'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o target -d 'Resource to target'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o var -d 'Set a variable in the Terraform configuration'
-complete -f -c terraform -n '__fish_seen_subcommand_from apply' -o var-file -d 'Set variables from a file'
+complete -f -c terraform -n "not __fish_terraform_needs_command" -o version -d "Print version information"
+complete -f -c terraform -o help -d "Show help"
+
+### apply/destroy
+set -l apply apply destroy
```

Comment:
would be easier to review if you placed this refactoring change (the consolidation of some complete-statements) in a separate commit.